R/Arena.R

```diff
@@ -2599,14 +2602,17 @@ setMethod("findFeeding3", "Eval", function(object, time, mets){
     if(class(interact)=="character"){interact = t(as.matrix(interact))}
     if(nrow(interact)!=0){inter = rbind(inter,data.frame(prod=interact[,1],cons=interact[,2],met=i))}
   }
-  if(any(dim(inter)==0)) stop("No feeding found. Try other metaboites or time points.")
+ if(any(dim(inter)==0)) {ifelse(test = (tail(times,n=1)==time),no= next, 
+ yes = print(paste("No feeding found. Try other metaboites or time points.")))}
+ else{          
   g <- igraph::graph.data.frame(inter[,1:2], directed=TRUE)
   l <- igraph::layout.kamada.kawai(g)
   plot(g,edge.color=grDevices::rainbow(length(levels(inter$met)))[as.numeric(inter$met)],
-       edge.width=3,edge.arrow.size=0.8,vertex.color=1:length(V(g)),layout=l)
-  legend("bottomright",legend=levels(inter$met),col=grDevices::rainbow(length(levels(inter$met))), pch=19, cex=0.7)
-  return(list(inter,g))
```

Comment:
To the point the discussion:

>  When in the function documentation it is written that a graph is returned you cannot just plot it.

Please explain to me what is wrong with this point. I have read the following page and to my understanding the command return is not useful for my approach,

> We generally use explicit return() functions to return a value immediately from a function.

> If it is not the last statement of the function, it will prematurely end the function bringing the control to the place from which it was called.

https://www.programiz.com/r-programming/return-function,